source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -21,11 +22,18 @@ public class JsonJournalRepository
         readonly ISemaphoreFactory semaphoreFactory;
         readonly string journalPath;
 
-        public JsonJournalRepository(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory)
+        public JsonJournalRepository(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory, IVariables variables)
         {
             this.fileSystem = fileSystem;
             this.semaphoreFactory = semaphoreFactory;
-            this.journalPath = @"C:\Octopus\PackageJournal.json";//journalPath;
+
+            var packageRetentionJournalPath = variables.Get(KnownVariables.Calamari.PackageRetentionJournalPath);
+            if (packageRetentionJournalPath == null)
+            {
+                var tentacleHome = variables.Get(TentacleVariables.Agent.TentacleHome) ?? throw new Exception("Environment variable 'TentacleHome' has not been set.");
```

Comment:
Other usages of `TentacleHome` throw an exception if it's null. Not sure what the alternative would be here.